app/src/main/java/com/keylesspalace/tusky/TabData.kt

```diff
@@ -98,6 +100,12 @@ fun createTabDataFromId(id: String, arguments: List<String> = emptyList()): TabD
             icon = R.drawable.ic_trending_up_24px,
             fragment = { TrendingTagsFragment.newInstance() }
         )
+        TRENDING_LINKS -> TabData(
```

Comment:
It would make sense to only have that "trending activity view with tabs" you'll get from the side drawer.

So the (only) tab here would show that as well.

(Consistent and removes clutter from this tab list, this is especially relevant for the following PR when this gets three tabs here.)

Reply:
That would put a set of tabs (trending tags, links, posts) inside another set of tabs.

That's possible (Material3 calls them "secondary tabs") but it's a departure from how we've done things so far. I also think it's reasonable for a user to only want one or two of them and not all three (e.g., I'd use links and posts, but have no interest in trending hashtags).

Reply:
Ok...
I see (secondary tabs).

But it would still be (quite) inconsistent:

- One location with a combined view (from side drawer)
- And one location where I would need to make a choice; and/or add 3 things seperately to get the same thing I get from the side drawer

:beard_scratch:

Reply:
@Lakoja Yes, so let's think about two typical users.

User A occasionally checks the trending info, but not habitually. Maybe a few times a week to see if there's anything new. They don't have a need to focus on one of the three (soon to be four) types of thing that can trend, so getting to it from the nav menu when they want to, and tapping through the different tabs to see what's new is fine for them.

User B is much more interested in trending links, they want to check regularly as part of their normal Mastodon experience. So much so that they want to make it a tab. They're not interested in posts or hashtags, just links. For them, adding a tab with just trending links is fine.

If they added a tab that contained all the trending items they would have to tap that tab once to choose "trending", and then tap again to choose which trending item they're interested in. That's suboptimal UX for an activity that User B is going to want to do a lot.

That's why I think the UX in this PR is the way to go.

There's a third user, User C, who are like User B, but they want to check all the trending tabs. They're not well served at the moment, because of the 5-tab limit, but I think that should be removed (see discussion in "Tusky General" channel).